### PR TITLE
Add compress model algorithm in Stackedlearner

### PR DESCRIPTION
**Pull request by hetong007**
As stated in this thread:https://github.com/mlr-org/mlr/issues/288#issuecomment-125407467, I have  implemented the model compression algorithm as a succeeding algorithm:

1. Run the hill climbing algorithm on a relatively small size of data
2. Generate some psuedo training data and label the data by the ensemble model from the hill climbing algorithm
3. Train a super learner (neural network by default) to mimic the behaviour of the ensemble. The result is this model, thus the size and the prediction efficiency of the model is reduced.

Since it relies on the hill climbing algorithm, I have implemented this algorithm in the same file. To make examples work, I also removed the `listLearners` command.

### CODE REVIEW

**Comment by larskotthoff** [comment]
Thanks, merging. It would be good to be able to pass the super learner as an argument.

**Comment by larskotthoff** [comment]
Wait, you've already implemented that. Ignore my comment :)

**Comment by hetong007** [comment]
Thank you Lars for reviewing my code :)
